.github/workflows/build_and_run_chain_simulator_and_execute_system_test.yml

```diff
@@ -179,7 +183,7 @@ jobs:
 
       - name: Upload test report
         if: always()
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```

Comment:
Why this change?

Reply:
It is required by GitHub since 30 January